test/functional/replace-by-fee.py

```diff
@@ -537,5 +541,29 @@ def test_rpc(self):
         assert_equal(json0["vin"][0]["sequence"], 4294967293)
         assert_equal(json1["vin"][0]["sequence"], 4294967294)
 
+    def test_mempool_replace_arg(self):
+        # ensure that no replacements are allowed with -mempoolreplacement=0
+        self.restart_node(0, ["-mempoolreplacement=0"])
+
+        tx0_outpoint = make_utxo(self.nodes[0], int(1.1*COIN))
+
+        tx1a = CTransaction()
+        tx1a.vin = [CTxIn(tx0_outpoint, nSequence=0)]
+        tx1a.vout = [CTxOut(1*COIN, CScript([b'a']))]
+        tx1a_hex = txToHex(tx1a)
+        tx1a_txid = self.nodes[0].sendrawtransaction(tx1a_hex, True)
+
+        # Extra 0.1 BTC fee double-spend that will fail due to argument being disabled
+        tx1b = CTransaction()
+        tx1b.vin = [CTxIn(tx0_outpoint, nSequence=0)]
+        tx1b.vout = [CTxOut(int(0.9*COIN), CScript([b'b']))]
+        tx1b_hex = txToHex(tx1b)
+        assert_raises_jsonrpc(-26, "txn-mempool-conflict", self.nodes[0].sendrawtransaction, tx1b_hex, True)
+        mempool = self.nodes[0].getrawmempool()
+
+        assert tx1a_txid in mempool
+
+        # We are not restarting the node here to save time, as this is the last test
```

Comment:
@jnewbery stop/start/restart could be deferred to the point they are needed unless a *flush* option is set. For instance:
```python
start_node(0)
stop_node(0)
```
would do nothing. This way tests could be more isolated.

Reply:
@promag interesting suggestion. I like the fact that it would make tests more isolated, but I don't like that it takes control of the nodes away from the test-writer. At the moment our tests are all very imperative, and this would be a slight departure from that model.

It's an interesting thought, but one that should be discussed in a separate issue or PR I think.

Reply:
> separate issue or PR I think.

Sure! Just a thought.